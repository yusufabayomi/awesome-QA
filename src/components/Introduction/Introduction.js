const Introduction = () => {
    return (
       <div className="text-center">
            <h1 data-test="appName" className="text-white f600">The Awesome QA Tool</h1>
            <p data-test="appDescription" className="text-white">Here you can find 0 questions. Feel free to create your own questions</p>
       </div>
    )
}

export default Introduction;