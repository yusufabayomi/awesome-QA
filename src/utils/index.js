export const required = (value) => (value ? undefined : 'Thid is a required field');

export const delay = (milliSeconds) => new Promise((res) => setTimeout(res, milliSeconds));

export const generateId = () => new Date().getTime().toString(36);

export const sortQuestions = (qas) => qas.sort((a, b) => a.question.localeCompare(b.question));
