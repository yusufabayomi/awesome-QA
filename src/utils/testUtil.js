
import checkPropTypes from 'check-prop-types';

export const findByTestAtrribute = (component, attr) => {
    const wrapper = component.find(`[data-test='${attr}']`);
    return wrapper;
};

export const checkProps = ({propTypes, name}, expectedProps) => {
    const propsErr = checkPropTypes(propTypes, expectedProps, 'props', name);
    return propsErr;
};