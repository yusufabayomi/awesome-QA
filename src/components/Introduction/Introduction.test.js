import React from 'react';
import { shallow } from 'enzyme';
import Introduction from './Introduction';
import { findByTestAtrribute } from '../../utils/testUtil';

describe('Renders Introduction Component', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Introduction />);
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Should render app name', () => {
    expect(findByTestAtrribute(wrapper, 'appName').length).toBe(1);
  });

  it('Should render app description', () => {
    expect(findByTestAtrribute(wrapper, 'appDescription').length).toBe(1);
  });
});
