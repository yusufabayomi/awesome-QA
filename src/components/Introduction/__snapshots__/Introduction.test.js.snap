// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Introduction Component renders correctly 1`] = `
<div
  className="text-center"
>
  <h1
    className="text-white f600"
    data-test="appName"
  >
    The Awesome QA Tool
  </h1>
  <p
    className="text-white"
    data-test="appDescription"
  >
    Here you can find 0 questions. Feel free to create your own questions
  </p>
</div>
`;
